src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Abstractions.Features;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
+{
+    internal class Http3Connection : IRequestProcessor
+    {
+        public HttpConnectionContext Context { get; private set; }
+
+        public DynamicTable DynamicTable { get; set; }
+
+        public Http3ControlStream SettingsStream { get; set; }
+        public Http3ControlStream EncoderStream { get; set; }
+        public Http3ControlStream DecoderStream { get; set; }
+
+        private readonly ConcurrentDictionary<long, Http3Stream> _streams = new ConcurrentDictionary<long, Http3Stream>();
+
+        // To be used by GO_AWAY
+        private long _highestOpenedStreamId; // TODO lock to access
+        //private volatile bool _haveSentGoAway;
+
+        public Http3Connection(HttpConnectionContext context)
+        {
+            Context = context;
+            DynamicTable = new DynamicTable(0);
+        }
+
+        internal long HighestStreamId
+        {
+            get
+            {
+                return _highestOpenedStreamId;
+            }
+            set
+            {
+                if (_highestOpenedStreamId < value)
+                {
+                    _highestOpenedStreamId = value;
+                }
+            }
+        }
+
+        public async Task ProcessRequestsAsync<TContext>(IHttpApplication<TContext> application)
+        {
+            var streamListenerFeature = Context.ConnectionFeatures.Get<IQuicStreamListenerFeature>();
+
+            // Start other three unidirectional streams here.
+            await CreateSettingsStream(application);
+            await CreateEncoderStream(application);
+            await CreateDecoderStream(application);
```

Comment:
Can/should these be done in parallel?

Reply:
Yes they can and yes they probably should.